### PR TITLE
Feature/pending candidates

### PR DESCRIPTION
**Pull request by jkleinekorte**
adds pending candidates in all botorch strategies

### CODE REVIEW

**Comment by jkleinekorte** [comment]
The test failed due to non-fitting of a model, which is caused by a badly sampled data as example. I will replace the sampling by a benchmark after the PR to remove the study is merged to main.

File: bofire/domain/features.py [hunk 1]
```diff
@@ -912,10 +905,11 @@ def get_bounds(
...
         else:
-            if values is None:
-                df = self.to_df().loc[self.get_allowed_categories()]
-            else:
-                df = self.to_df().loc[self.get_possible_categories(values)]
+            df = self.to_df()
+            # if values is None:
+            #     df = self.to_df().loc[self.categories]
+            # else:
+            #     df = self.to_df().loc[self.get_possible_categories(values)]
```
Comment (jduerholt) [inline bofire/domain/features.py:912]: I think we have to comment this in again. Let us talk later about it.

File: bofire/domain/features.py [hunk 2]
```diff
@@ -787,15 +787,8 @@ def get_bounds(
...
-                lower = [0.0 for _ in self.categories]
-                upper = [
-                    1.0 if self.allowed[i] is True else 0.0  # type: ignore
-                    for i, _ in enumerate(self.categories)
-                ]
-            else:
-                lower = [0.0 for _ in self.categories]
-                upper = [1.0 for _ in self.categories]
+            lower = [0.0 for _ in self.categories]
+            upper = [1.0 for _ in self.categories]
```
Comment (jduerholt) [inline bofire/domain/features.py:791]: Revert.

File: bofire/domain/features.py [hunk 3]
```diff
@@ -912,10 +905,11 @@ def get_bounds(
         if transform_type != CategoricalEncodingEnum.DESCRIPTOR:
             return super().get_bounds(transform_type, values)
         else:
-            if values is None:
-                df = self.to_df().loc[self.get_allowed_categories()]
```
Comment (jduerholt) [inline bofire/domain/features.py:916]: this line is correct.

File: bofire/domain/features.py [hunk 4]
```diff
@@ -912,10 +905,11 @@ def get_bounds(
...
-            else:
-                df = self.to_df().loc[self.get_possible_categories(values)]
```
Comment (jduerholt) [inline bofire/domain/features.py:918]: this line is wrong

**Review by jkleinekorte — COMMENTED** [review]
Changes are fine for me!

**Review by jduerholt — APPROVED** [review]
LGTM.

**Pull request merged**